sktime/utils/_testing/estimator_checks.py

```diff
@@ -430,6 +521,36 @@ def check_methods_do_not_change_state(Estimator):
             ), f"Estimator: {estimator} changes __dict__ during {method}"
 
 
+def check_methods_have_no_side_effects(Estimator):
+    # Check that calling methods has no side effects on args
+
+    if not isclass(Estimator):
+        Estimator = type(Estimator)
+
+    estimator = _construct_instance(Estimator)
+
+    set_random_state(estimator)
+
+    # Fit for the first time
+    fit_args = _make_args(estimator, "fit")
+    old_fit_args = deepcopy(fit_args)
+    estimator.fit(*fit_args)
```

Comment:
@fkiraly, I think the reason why the detrender was not caught is here:  `_make_args(estimator, "fit")` creates a `pd.Series`. But as side effects are only introduced in the multivariate case, they won't be caught.

Reply:
ah! Well observed, I didn't notice this.

So, is there a potential action on adding tests for multivariate arguments to `fit`?
Perhaps that makes sense only once we've settled on a "can handle multivariate" tag, because only if the estimator claims it can an error should be thrown.

Reply:
In theory, you could already add this to the tests, as we have the `univariate-only tag` -  At least I would assume that each transformer should be able to handle multivariate data, unless it carries the `univariate-only` tag.

Reply:
ahhhhh but that is not the case, that would be too easy, @SveaMeyer13!
This is exactly what I thought, but according to one of @mloning and @aiwalter (I forgot which) the tag is actually meant to signify whether the exogeneous variables `X` are used or ignored - not whether `y` can be a data frame.